### PR TITLE
Updating `maven` to version 3.8.1 to fix CVE-2021-26291.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge
---

###### Summary 

Updating `maven` to version 3.8.1 to fix CVE-2021-26291.

###### Change Log  

- Updating `maven` to version 3.8.1.
- Added a conditionally built `maven-cached-sources` subpackage to simplify future updates.

###### Does this affect the toolchain?  

No.

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-26291

###### Test Methodology

- Local package build.

### CODE REVIEW

File: SPECS/maven/maven.spec [hunk 1]
```diff
@@ -1,65 +1,109 @@
+# Switch "sources_generation" to 1 when running a package build to generate cached sources for regular builds.
+%define sources_generation 0
+%define m2_cache_tarball_name apache-%{name}-%{version}-m2.tar.gz
+%define licenses_tarball_name apache-%{name}-%{version}-licenses.tar.gz
+
+%if ! 0%{?sources_generation}
+%define offline_build -o
```
Comment (jslobodzian) [inline SPECS/maven/maven.spec:7]: Seems like we ALWAYS want that set.  doesn't hurt to have the option here, but you could reduce the complexity.

File: SPECS/maven/maven.spec [hunk 2]
```diff
@@ -1,65 +1,109 @@
...
 License:        ASL 2.0
-URL:            https://maven.apache.org/
-Group:          Applications/System
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
+Group:          Applications/System
+URL:            https://maven.apache.org/
 Source0:        https://archive.apache.org/dist/maven/maven-3/%{version}/source/apache-%{name}-%{version}-src.tar.gz
+# Using pre-compiled binaries because 'maven' requires itself during build-time.
 Source1:        https://archive.apache.org/dist/maven/maven-3/%{version}/binaries/apache-%{name}-%{version}-bin.tar.gz
```
Comment (jslobodzian) [inline SPECS/maven/maven.spec:21]: We should take a bug on this to use our OWN binaries from this build once it completes.

File: SPECS/maven/maven.spec [hunk 3]
```diff
@@ -1,65 +1,109 @@
...
+# and make sure network is enabled during the build. The tarballs will be inside the built 'maven-cached-sources' subpackage.
+%if ! 0%{?sources_generation}
+Source2:        %{m2_cache_tarball_name}
+Source3:        %{licenses_tarball_name}
+%endif
 BuildRequires:  ant
-BuildRequires:  openjre8
 BuildRequires:  openjdk8
+BuildRequires:  openjre8
 BuildRequires:  wget >= 1.15
```
Comment (jslobodzian) [inline SPECS/maven/maven.spec:31]: not in scope..but doe we even need that 1.15 qualifier?

**Review by jslobodzian — APPROVED** [review]

File: SPECS/maven/maven.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/maven/maven.spec:7]: We don't want to have this set when `sources_generation` is equal to 1. This option makes Maven build in offline mode and we need it to use the network when we generate the tarballs.

File: SPECS/maven/maven.spec (hunk 2 above)
Comment (PawelWMS) [reply SPECS/maven/maven.spec:21]: That's part of a separate topic I'm planning to discuss tomorrow.

File: SPECS/maven/maven.spec (hunk 3 above)
Comment (PawelWMS) [reply SPECS/maven/maven.spec:31]: The build will be fine without it, but having additional information about the lower bound on build-time dependencies is a nice thing to have. Whether this particular line is currently true, however, is another thing.

File: SPECS/maven/maven.spec (hunk 2 above)
Comment (PawelWMS) [reply SPECS/maven/maven.spec:21]: Bug created.

**Pull request merged**